src/bin/main.rs

```diff
@@ -62,8 +62,7 @@ enum SubCommand {
 
 fn main() -> anyhow::Result<()> {
     let opts: Opts = Opts::parse();
-    let keep_name_section = matches!(opts.subcommand, SubCommand::Shrink { .. });
```

Comment:
Unless I'm mistaken, this logic preserves the name section when using the `shrink` command, which is not the intended behavior?

And here, due to walrus's behavior, the name section is not deleted either:
https://github.com/dfinity/ic-wasm/blob/3246f6d4e888064a3895e2cd88853727933c187d/src/shrink.rs#L20